### PR TITLE
[DSM] DDP-6584 downloading participant files

### PR DESCRIPTION
**Pull request by pegahtah**
An endpoint `DownloadParticipantFileRout`  is added to handle the downloading requests. 
The user should have `file_download` access role to be able to do the download
ElasticSearchUtil has also been changed so users are able to search for files and sort on the time

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = null;
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        }
+        if (StringUtils.isBlank(realm)) {
+            response.status(500);
+            throw new RuntimeException("Realm was missing from the request");
+        }
+        String userIdR = UserUtil.getUserId(request);
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: I think we already have `userId` in the arguments of `processRequest`?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            String objectName = queryParams.value("fileName");
+            String blobName = queryParams.value("blob");
+            Optional<byte[]> downloadObject =
+                    downloadObject(googleProjectName, bucketName, objectName, blobName);
+            byte[] downloadObjectArray = downloadObject.orElseThrow();
+            HttpServletResponse rawResponse = response.raw();
+            rawResponse.getOutputStream().write(downloadObjectArray);
+            rawResponse.setStatus(200);
+            rawResponse.getOutputStream().flush();
+            rawResponse.getOutputStream().close();
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: why code from line 50 to line 54 is necessary? Just returning `downloadObjectArray` is not enough?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            return new Result(200);
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: As we agreed, step by step we will try to remove returning `Result` object from routes, let's just return empty string if route is void.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        } else {
+            return new Result(401, SecurityUtil.ResultType.AUTHENTICATION_ERROR.toString());
```
Comment (gmakharat) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: response.status(401);
return SecurityUtil.ResultType.AUTHENTICATION_ERROR.toString();

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java (hunk 1 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: We check that these two match in `UserUtil.checkUserAccess`

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java (hunk 3 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: True, I will fix that, thanks for the reminder!

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java (hunk 2 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java]: It is enough. fixed

**Review by gmakharat — APPROVED** [review]

**Pull request merged**